### ISSUE TITLE
Create stack template(s) for beginners

### ISSUE THREAD

**Issue by MatthijsBlom** [tomjaguarpaw/tilapia#65]
Stack has a few thinks going for it:

-   Installation of GHC &c. _just works_ (for beginners, anyway);
-   The viability of stack as a setup tool need not be maintained by tilapia, as it is already being maintained by commercialhaskell;
-   At some point the advanced beginner will need to interact with a build tool anyway &ndash; might as well be this one;
-   **Stack templates offer ample space for explanatory material.** With only a few exceptions (among them, sadly, `stack.yaml`), every file is fully customizable, and could therefore be supplied with notes tailored to beginners.

(Does Cabal have similar functionality?)

I propose a series of templates be produced, from, say, a minimal executable, to a simple library, to more typical projects, to a grand tour of stack exotica.

I'm curious to hear what you think.

---

See [commercialhaskell/stack-templates](https://github.com/commercialhaskell/stack-templates#readme) for basic instructions.

The Stack user guide [suggests](https://docs.haskellstack.org/en/stable/GUIDE/#templates) publishing of templates is very easy; more difficult will be making sure people know they exist &ndash; and writing them, of course.

**Comment by tomjaguarpaw** [comment]
Oh, very interesting idea. I don't use Stack and I had no idea that these templates existed! I've never heard of anything like this for cabal. Do the templates have deep integration with Stack or would some lightweight mustache-like templactes do a similar job for cabal?

**Comment by MatthijsBlom** [comment]
> [&ndash;] Do the templates have deep integration with Stack 

Well, templates are inherently single-use per project, but initialization of stack projects with `stack new` always use a template. Trying it for yourself is very easy: just install stack and run `stack new ProjectName`. You can compare the result with the [source of the default template](https://github.com/commercialhaskell/stack-templates/blob/master/new-template.hsfiles). See also the [relevant section](https://docs.haskellstack.org/en/stable/GUIDE/#templates) of the Stack user guide.

> or would some lightweight mustache-like templactes do a similar job for cabal?

I wouldn't know, as I do not yet know Mustache.

**Comment by MatthijsBlom** [comment]
> > [&hellip;] Do the templates have deep integration with Stack
> 
> Well, templates are inherently single-use per project, but initialization of stack projects with `stack new` always use a template. Trying it for yourself is very easy: just install stack and run `stack new ProjectName`. You can compare the result with the [source of the default template](https://github.com/commercialhaskell/stack-templates/blob/master/new-template.hsfiles). See also the [relevant section](https://docs.haskellstack.org/en/stable/GUIDE/#templates) of the Stack user guide.
> 
> > or would some lightweight mustache-like templactes do a similar job for cabal?
> 
> I wouldn't know, as I do not yet know Mustache.

**Comment by tomjaguarpaw** [comment]
Ah, I see. It sounds like you are suggesting creating templates that a new user can use like `stack new myproject tilapia/minimal-executable` or `stack new myproject tilapia/simple-library`. Is that right?

**Comment by MatthijsBlom** [comment]
That's right.

**Comment by tomjaguarpaw** [comment]
Yes, I like that idea. Cabal doesn't have anything similar as far as I know.

**Comment by ketzacoatl** [comment]
How would these new templates differ from https://github.com/commercialhaskell/stack-templates/blob/master/simple-library.hsfiles and https://github.com/commercialhaskell/stack-templates/blob/master/simple.hsfiles, or are those sufficiently useful as-is?
